tests/conftest.py

```diff
@@ -396,3 +397,98 @@ def log_cli_level(pytestconfig):
 
     """
     return pytestconfig.getini('log_cli_level') or 'DEBUG'
+
+
+@pytest.fixture(scope="class")
+def run_io_in_background(request):
+    """
+    Run IO during the test execution
+    """
+    if config.RUN['cli_params'].get('io_in_bg'):
+
+        lock = threading.Lock()
+        with lock:
+
+            # test_running is used for communication between the main thread
+            # and the new thread used for running the IO.
+            # Keeping test_running in a list because bool is immutable.
+            # Locking is used here as a synchronization mechanism to guard
+            # access to test_running.
+            request.test_running = [True]
+
+        results = list()
+
+        def finalizer():
+            """
+            Delete the resources created during setup, used for
+            running IO in the test background
+            """
+            with lock:
+                request.test_running = [False]
+            thread.join()
+            for result in results:
+                log.info(f"IOPs after FIO for pod {request.pod_obj.name}:")
+                log.info(
+                    f"Read: {result.get('jobs')[0].get('read').get('iops')}"
+                )
+                log.info(
+                    f"Write: {result.get('jobs')[0].get('write').get('iops')}"
+                )
+            if hasattr(request, 'pod_obj'):
+                request.pod_obj.delete()
+                request.pod_obj.ocp.wait_for_delete(
+                    resource_name=request.pod_obj.name
+                )
+            if hasattr(request, 'pvc_obj'):
+                request.pvc_obj.delete()
+                request.pvc_obj.ocp.wait_for_delete(
+                    resource_name=request.pvc_obj.name
+                )
+            if hasattr(request, 'sc_obj'):
+                request.sc_obj.delete()
+            if hasattr(request, 'secret_obj'):
+                request.secret_obj.delete()
+
+        request.addfinalizer(finalizer)
+
+        request.secret_obj = helpers.create_secret(
```

Comment:
Can we utilize fixtures for these dependencies instead of calling the helper functions directly? They would need to be class (or larger) scoped fixtures for the secret, storage_class, pvc, and pod. I think this could help clean this fixture up a bit since we wouldn't have to handle creation/deletion here.

Reply:
this means we'll have to move all these fixtures from fixtures.py to conftest and adjust many tests to it.
I suggest we wait for https://github.com/red-hat-storage/ocs-ci/pull/418 to get merged as it already covers this. Then we'll make the necessary adjustments in this new fixture

Reply:
@ebenahar that works for me!